applications/party/groovyScripts/party/PartyServices.groovy

```diff
@@ -427,13 +427,49 @@ def createPartyRelationship() {
                partyIdTo: parameters.partyIdTo,
                roleTypeIdFrom: parameters.roleTypeIdFrom,
                roleTypeIdTo: parameters.roleTypeIdTo)
-        .filterByDate()
         .queryList()
-    if (!partyRels) {
-        GenericValue partyRelationship = makeValue('PartyRelationship', parameters)
-        partyRelationship.create()
+
+    if (partyRels) {
+        // checks for conflicts with existing unlimited valid entries
+        for (GenericValue partyRel : partyRels) {
+            if (!partyRel.thruDate) {
+                if (parameters.fromDate.after(partyRel.fromDate))
+                    return success()
+            }
+        }
+
+        // checks if there are any conflicts with the time periods of the entries
+        if (!parameters.thruDate) {
+            for(GenericValue partyRel : partyRels) {
+                if ((parameters.fromDate.before(partyRel.fromDate)) || (parameters.fromDate.before(partyRel.thruDate))) {
+                    return success()
+                }
+            }
+        } else {
+            EntityCondition partyCond =
```

Comment:
Instead of having logic in createPartyRelationship service, we can make it simple create service, service will simply create party relationship record, and at the business side we can implement checks

Reply:
Hi Deepak, this change is explixitely provided to enhance the business logic on this central point so that code using it does not have to deal with it.

Reply:
@dixitdeepak are you ok with the change after my explanation?